ownership/views.py

```diff
@@ -54,11 +55,14 @@ def _respond(self):
     def get_context_data(self, **kwargs):
         context = {}
         delivery = self.delivery
-        if delivery:
-            details = self.delivery_type.make_delivery_details(delivery, self.request.user)
-            context.update(details.get_context())
-            context['delivery_type'] = self.delivery_type.name
-        return context
+        try:
+            if delivery:
+                details = self.delivery_type.make_delivery_details(delivery, self.request.user)
+                context.update(details.get_context())
+                context['delivery_type'] = self.delivery_type.name
+            return context
+        except S3NotRecipientException:
+            self.set_error_details(401, NOT_RECIPIENT_MSG)
```

Comment:
I agree the word "unauthorized" fits the state here: _You, as the sender, do not have the **authority** to accept this delivery_, but don't think 401 is the right status:

From https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
> The HTTP 401 Unauthorized client error status response code indicates that the request has not been applied because it lacks valid authentication credentials for the target resource.
> This status is sent with a WWW-Authenticate header that contains information on how to authorize correctly.

Since we're not really asking the client to try again with a different password, I think 403 is a better fit:

From https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403
> The HTTP 403 Forbidden client error status response code indicates that the server understood the request but refuses to authorize it.

HTTP 403 Forbidden: The server refuses to authorize it. 🤦‍♂️ 

Also, http://www.dirv.me/blog/2011/07/18/understanding-403-forbidden/index.html